File: 프로그래머스/2/42626. 더 맵게/더 맵게.py
import heapq

def solution(scoville, K):
    answer = 0
    
    heapq.heapify(scoville)
    
    while len(scoville) >= 2:
        first = heapq.heappop(scoville)
        if first >= K:
            return answer
        
        second = heapq.heappop(scoville)
        new_scoville = first + (second * 2)
        heapq.heappush(scoville, new_scoville)
        answer += 1
        
    return answer if scoville[0] >= K else -1

File: 프로그래머스/2/42626. 더 맵게/더 맵게.java
import java.util.*;

class Solution {
    public int solution(int[] scoville, int K) {
        int answer = 0;
        
        var pq = new PriorityQueue<Integer>();
        for (int s: scoville) {
            pq.add(s);
        }
        
        while (pq.size() != 1) {
            if (pq.peek() >= K) {
                return answer;
            }
            
            pq.add(pq.poll() + (pq.poll() * 2));
            answer += 1;
        }
        
        if (pq.peek() >= K) {
            return answer;
        }
        return -1;
    }
}